app/controllers/api/v0/articles_controller.rb

```diff
@@ -4,7 +4,7 @@ class ArticlesController < ApiController
       respond_to :json
 
       before_action :authenticate!, only: %i[create update me]
-
+      before_action -> { doorkeeper_authorize! :public }, only: %w[index show me], if: -> { doorkeeper_token }
```

Comment:
`me` doesn't seem like a `public` scope to me because it contains unpublished articles. Or are they "public" to a specific user? I would split this action to separate ones to be able to authorize accordingly.

Reply:
I think it should be kept private. Looking at how GitHub does this, they have a `repo` check box which grants all public + private repo, and a `public_repo` which doesn't grant access to private repos.

Reply:
Can't we just check which scope they've authorized to decide if we should send them unpublished articles or not?

Reply:
@rhymes do you mean when sending events? Currently, the question is about authorizing the API requests.
I think, sending events based on what the user had authorized is possible now (after connecting webhooks and oauth apps), but for the mvp we agreed to send only the events about published articles.

Reply:
@lightalloy I mean resource access in general. `/me/unpublished` should not be accessible if the OAuth is only public, that's why I was thinking about using [Pundit policies](https://github.com/varvet/pundit#scopes) (or a manual mechanism) because you can easily check what the user can access to, and if they are authorized only for published posts, then `/unpublished` will return an empty list for example.

Reply:
So, to achieve it easier we should split `me` action to actions that display `published` and `unpublished` articles. That way we can use doorkeeper to authorize. I don't think it makes sense to add pundit policies to it if we authorize based on the doorkeeper scopes.

Reply:
Maybe you can try calling the `doorkeeper_authorize!` method at runtime before splitting